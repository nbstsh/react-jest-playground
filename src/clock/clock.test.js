import React from 'react';
import Clock from './clock.components';
import renderer from 'react-test-renderer';

describe('clock component', () => {
	window.Date = function() {
		this.toISOString = jest.fn().mockReturnValue('123456789');
	};

	it('should render clock.', () => {
		const component = renderer.create(<Clock />);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();

		console.log(JSON.stringify(tree, null, 2));
	});
});

// test('Link changes the class when hovered', () => {
// 	const component = renderer.create(
// 		<Link page='http://www.facebook.com'>Facebook</Link>
// 	);
// 	let tree = component.toJSON();
// 	expect(tree).toMatchSnapshot();

// 	// manually trigger the callback
// 	tree.props.onMouseEnter();
// 	// re-rendering
// 	tree = component.toJSON();
// 	expect(tree).toMatchSnapshot();

// 	// manually trigger the callback
// 	tree.props.onMouseLeave();
// 	// re-rendering
// 	tree = component.toJSON();
// 	expect(tree).toMatchSnapshot();
// });
