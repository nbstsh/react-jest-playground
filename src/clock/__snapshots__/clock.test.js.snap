// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clock component should render clock. 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#6b6eff",
      "borderRadius": "100px",
      "boxShadow": "1px 1px 3px 1px #0000003b",
      "color": "#fff",
      "display": "flex",
      "fontSize": "20px",
      "fontWeight": "bold",
      "justifyContent": "center",
      "maxWidth": "400px",
      "padding": "20px 30px",
    }
  }
>
  <p>
    123456789
  </p>
</div>
`;
